modules/input.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.plasma;
+  numlockSettings = [ "on" "off" "unchanged" ];
+in
+{
+  # Keyboard options
+  options.programs.plasma.input.keyboard = {
+    layouts = mkOption {
+      type = with types; nullOr (listOf str);
+      default = null;
+      example = [ "es" "us" ];
+      description = ''
+        Keyboard layouts to use.
+      '';
+    };
+    numlockOnStartup = mkOption {
+      type = with types; nullOr (enum numlockSettings);
+      default = null;
+      example = "on";
+      description = ''
+        Numpad settings at startup.
+      '';
+    };
+  };
+
+  config.programs.plasma.configFile."kxkbrc" = mkIf (cfg.enable) (
+    mkMerge [
+      (
+        mkIf (cfg.input.keyboard.layouts != null) {
+          Layout = {
+            Use.value = true;
+            LayoutList.value = strings.concatStringsSep "," cfg.input.keyboard.layouts;
+          };
+        }
+      )
+    ]
+  );
+
+  # Trackpad options
+
+  options.programs.plasma.input.touchpad = {
+    enable = mkOption {
+      type = with types; nullOr bool;
+      default = null;
+      example = true;
+      description = ''
+        Enables or disables the trackpad
+      '';
+    };
+    name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "PNP0C50:00 0911:5288 Touchpad";
+      description = ''
+        The name of the trackpad.
+        You can find it out running cat /proc/bus/input/devices | grep -i touchpad
+      '';
+    };
+    vendorId = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "2321";
+      description = ''
+        The vendor ID of the trackpad
+        You can find it out running cat /proc/bus/input/devices | grep -i touchpad
+      '';
+    };
+    productId = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "21128";
+      description = ''
+        The product ID of the trackpad
+        You can find it out running cat /proc/bus/input/devices | grep -i touchpad
+      '';
+    };
```

Comment:
These three have allowed values as `null` but if the touchpad is enabled and these are not set I'm pretty sure it will crash. This should probably be checked in an assertion so we could give out a proper error message.